feat: Legend visibility, one-line hover, fixed merge conflicts

myosotis-rain opened the pull request.

myosotis-rain committed:
Panel styling updates (prior to newest design - some changes needed)

- Typography: H1 20px semibold, H2 16px semibold, Inter font
- Panel positioning anchored from Clear All Layers button
- 10px gaps between panels
- 24px chevron icons
- Dropdown with slate-200 background and custom arrows
- Blue-600 checkbox colors
- Moved inline JS/CSS to external files
- Centered legend content
- Updated text casing and weights

myosotis-rain committed:
Fix fonts and update layout for Area dropdown and Clear Layers

- Update Area Feature dropdown font to Inter
- Fix Clear Layers button font styling
- Improve flex container spacing with 10px gap
- Make dropdown responsive with flex: 1 and width: 100%
- Update More button padding to 0.5rem

myosotis-rain committed:
Added hover highlight feature to the body titles

myosotis-rain committed:
Fix checkbox alignment and styling

- Align checkboxes with subheading text by adjusting margin-left to 1.5rem
- Update checkbox colors from #2563eb to #155DFC
- Add white checkmark for checked state
- Change display from inline-flex to flex

myosotis-rain committed:
Enable full-row clicking for panel and section toggles

myosotis-rain committed:
Hide legend until content is added and fix title positioning

- Hide legend by default until attributes/layers are added
- Move legend title and chevron to top when content is present
- Add has-content class management in updateLegendWidth function
- Ensure legend stays hidden even in closed state when no content

myosotis-rain committed:
Merge remote-tracking branch 'origin/main' into legend_visibility_feat

myosotis-rain committed:
Restore panel styling and functionality after merge conflicts

- Fix panel toggle functionality for full-row clicking
- Implement consistent spacing: 0.75rem from panel edges, 0.5rem between elements
- Add proper hover effects for checkbox items with full-width background
- Fix checkbox text wrapping and More button alignment
- Standardize chevron positioning and styling (1.5rem x 1.5rem, 0.25rem radius)
- Ensure consistent top/left spacing across Public Data and Uploaded Data panels
- Reduce spacing conflicts and overlapping margins/padding
- Improve Select2 containment and styling within panels

danikam commented:
Hi @myosotis-rain, thanks for the PR and your work on this! Please find my comments below, and don't hesitate to let me know in case something is unclear.

**A few items to please address before we merge this in:**
* It looks like there's been some vertical space added between the Legend header and the first legend entry (it's currently ~60px / 3.75rem, and should be 12px / 0.75 rem per the Figma design).

<img width="530" height="220" alt="Screenshot 2025-10-17 at 5 36 29 PM" src="https://github.com/user-attachments/assets/5731816b-18b6-4793-8c96-18c0c76ef3b6" />

* As we discussed during our meeting today, the collapsible section headers (eg. 'Highway Flows', 'Planned Infrastructure Corridors', etc.) should also get highlighted in light blue on hover.
* It looks like the chevron in the public data and uploaded layers boxes got a bit shifted to the left for some reason. Based on the Figma design, they should be 12px / 0.75rem to the left of the border, and they're currently  ~24px / 1.5rem to the left.

<img width="1129" height="352" alt="Screenshot 2025-10-17 at 6 04 26 PM" src="https://github.com/user-attachments/assets/ea65929b-cae5-4cf3-b0c6-d22b46919e90" />

* The blue More button that appears on hover looks to have been moved quite a bit to the left (see attached). Per the Figma design, the right side of the More button should be 12px / 0.75rem to the left of the box border.

Current appearance:
<img width="276" height="73" alt="Screenshot 2025-10-17 at 6 07 35 PM" src="https://github.com/user-attachments/assets/d52cf5c5-88df-44b6-9492-087998d32375" />

Figma design:
<img width="750" height="372" alt="Screenshot 2025-10-17 at 6 07 27 PM" src="https://github.com/user-attachments/assets/ac8362c3-0255-4305-a19e-c920f1ec7b0d" />

* The blue highlight on hover should extend across the full width of the box (currently there's a bit of white space to the left and right)
<img width="276" height="73" alt="Screenshot 2025-10-17 at 6 07 35 PM" src="https://github.com/user-attachments/assets/acf89f11-fa7e-478d-8654-9f320d1e4ec9" />

* The horizontal space allotted for text in the Public Data checkboxes is getting slightly reduced on hover, which sometimes causes labels to change from one to two lines on hover. The space allotted for text should stay the same regardless of whether the user is hovering over the checkbox line or not. 
<img width="249" height="35" alt="Screenshot 2025-10-17 at 6 13 41 PM" src="https://github.com/user-attachments/assets/01f18f76-5752-43dd-a79f-c97cbd4d89ed" />
<img width="266" height="57" alt="Screenshot 2025-10-17 at 6 13 51 PM" src="https://github.com/user-attachments/assets/c1df0ab3-43c4-41ea-ba63-24fa44d9ac94" />

* The checkbox labels have been shifted up a bit relative to the checkboxes. Please ensure that the checkbox labels are vertically aligned with the checkbox.

Current appearance:
<img width="522" height="88" alt="Screenshot 2025-10-17 at 6 14 45 PM" src="https://github.com/user-attachments/assets/e852ccd0-36bf-4d24-a186-f90099b3f16d" />

Figma design:
<img width="160" height="68" alt="Screenshot 2025-10-17 at 6 16 06 PM" src="https://github.com/user-attachments/assets/14e5e670-c948-41d5-ba09-c56869240f20" />


**To be addressed down the line (not needed to merge this PR):**
* We can discuss with Viktoriia whether it makes sense to keep the legend invisible before any layers are selected, but fine to keep it as-is for now.

myosotis-rain commented:
<img width="282" height="569" alt="Screenshot 2025-10-21 at 11 43 40 AM" src="https://github.com/user-attachments/assets/899124f1-e8a1-4f8f-bcba-71429b8cd7f0" />
Hi Danika! That's quite odd because the more button looks like this on my screen. I'll inspect and try to see if editing the placement would help with that. Thank you!

myosotis-rain committed:
fix: fixed hovering functionality and some spaces; legend still having extra space

danikam commented:
Hmm, that's interesting about the differing appearance. For reference, I was viewing it in Firefox. Happy to try adjusting the More button placement on my side in case you're still seeing it closer to the box border - just let me know!

myosotis-rain committed:
fix: improve UI positioning and legend visibility

- Fix chevron positioning in panel headers with subtle adjustment
- Improve text alignment in Select2 placeholder with negative margins
- Adjust checkbox alignment to match first line of text with margin-top
- Fix panel header margins for consistent left alignment
- Increase uploaded data panel closed state width for better fit
- Ensure legend defaults to open state when content first appears
- Reset height constraints when opening closed legend

myosotis-rain committed:
update: change Clear Layers button text to Clear all layers

danikam commented:
Hi @myosotis-rain, thanks a lot for your updates, great work! These address all the items I noted in my initial review. 

Regarding the other updates you outlined on slack:
* The current width of the Public Data box looks good to me (and I think it's ok that ‘National ZEF Corridor Strategy’ gets split into two lines, as you noted on slack). 
* The vertical centering of the ‘More’ button looks good and was a great choice. 

Just two other items I noticed while reviewing (I'm not sure if these were introduced in your last two commits or I just missed them in my initial review, apologies if it's the latter!):
1. Only the first option is appearing as a bullet point under the area dropdown even if more than one is available/modified. For example, looking at the 'Lifecycle Truck Emissions' layer, only the 'Average payload: []' bullet appears, even if one of the other two options are modified:
<img width="272" height="109" alt="Screenshot 2025-10-29 at 1 40 11 PM" src="https://github.com/user-attachments/assets/93c83418-b43a-499b-94d6-bf5c0970f933" />
<img width="751" height="387" alt="Screenshot 2025-10-29 at 1 40 42 PM" src="https://github.com/user-attachments/assets/1df00de7-55d2-4b8a-984f-fc33e45eb361" />


This looks to still be ok for the ‘Savings from Pooled Infrastructure’ checkbox (all three bullets are visible), so only the area dropdown needs to be addressed.

2. Please align bullets under the ‘Savings from Pooled Infrastructure’ checkbox near the right of the checkbox, as in the Figma.

Current: 
<img width="271" height="119" alt="Screenshot 2025-10-29 at 1 25 13 PM" src="https://github.com/user-attachments/assets/5be7194d-dc12-4778-8116-2560f815ef02" />

Figma:
<img width="242" height="151" alt="Screenshot 2025-10-29 at 1 25 21 PM" src="https://github.com/user-attachments/assets/04d14f82-40a7-434e-85a0-75a448a3cf21" />

Could you please address the above two items before we merge this in?

Thanks!
-Danika

myosotis-rain committed:
fix: point features attributes & area attributes location + display

danikam commented:
Spacing of bullets looks great. Please align the behaviour of the attribute bullets so they're the same between all area dropdown layers. Then should at last be ready to merge in!

Thanks for your awesome work @myosotis-rain!

myosotis-rain committed:
refactor: modularize codebase, clean up code, and fix attributes display

- Extract layer management utilities into layer-utils.js
- Create bullet-config.js for parameter configuration
- Add transform-utils.js for reusable transform functions
- Simplify map.js by removing ~300 lines of duplicated code
- Clean up HTML template comments and duplicate imports
- Fix attributes display functionality

danikam commented:
Thanks so much @myosotis-rain for these updates! The new automated bullet-adding function looks and works great, and the placement of the bullets looks perfect. 

I noticed an issue in the uploaded data box, which looks to have been introduced in [this commit](https://github.com/mcsc-impact-climate/Geo-TIDE/commit/d5c900eb5325482a6831f9090bd7fad334dd0da5) that came immediately after [merging in updates from main](https://github.com/mcsc-impact-climate/Geo-TIDE/commit/fb2b46463901a6825f6d8f7a74ff68d4f5c64e70).

Here's the appearance from the [merge commit](https://github.com/mcsc-impact-climate/Geo-TIDE/commit/fb2b46463901a6825f6d8f7a74ff68d4f5c64e70):
<img width="273" height="155" alt="Screenshot 2025-11-03 at 5 27 57 PM" src="https://github.com/user-attachments/assets/ca284586-3d26-4345-b694-a3e994414173" />

Subsequent commit (selected layers don't appear at all):
<img width="271" height="98" alt="Screenshot 2025-11-03 at 5 29 06 PM" src="https://github.com/user-attachments/assets/a90548cb-8ee6-4324-b96a-5a924d907707" />

And most recent commit (pillboxes for selected layers appear but are cropped and all on one line):
<img width="303" height="99" alt="Screenshot 2025-11-03 at 5 31 04 PM" src="https://github.com/user-attachments/assets/4cb0b21b-3591-471d-8d6c-e09f4f2025fb" />

I'm guessing this was just a side-effect of some other UI update and I should have been testing for it more carefully (especially since I assume you don't currently have uploaded layers on your side to test with). 

Since I haven't yet shown you how to add uploaded layers and test their appearance in the Uploaded Data box, I'll merge this in for now and make a to-do for myself to the cropped pillboxes in a separate branch. Happy to get you set up with some uploaded layers next time we meet if that would be helpful!

Thanks again for all your awesome work on this!

danikam commented:
@myosotis-rain, the `legend_visibility_feat` branch has now been merged into `main` and deleted! Please make sure to do the following on your local repository to help keep things clean moving forward:

* Update your main branch and remove the `legend_visibility_feat` (which no longer exists on the remote repo):

```bash
git checkout main
git pull
# Assuming there's no uncommitted work on your local legend_visibility_feat, you're now safe to delete the branch locally
git branch -D legend_visibility_feat
```

* Make sure to merge updates from main into any other branches you're working on:

```bash
git checkout your-feature-branch
git fetch origin
git merge origin/main
```

and resolve any merge conflicts.

Thanks!
